src/Link.cc

```diff
@@ -108,6 +112,71 @@ std::optional<Model> Link::ParentModel(const EntityComponentManager &_ecm) const
   return std::optional<Model>(parent->Data());
 }
 
+//////////////////////////////////////////////////
+Entity Link::CollisionByName(const EntityComponentManager &_ecm,
+    const std::string &_name) const
+{
+  return _ecm.EntityByComponents(
+      components::ParentEntity(this->dataPtr->id),
+      components::Name(_name),
+      components::Collision());
+}
+
+//////////////////////////////////////////////////
+Entity Link::VisualByName(const EntityComponentManager &_ecm,
+    const std::string &_name) const
+{
+  return _ecm.EntityByComponents(
+      components::ParentEntity(this->dataPtr->id),
+      components::Name(_name),
+      components::Visual());
+}
+
+//////////////////////////////////////////////////
+std::vector<Entity> Link::Collisions(const EntityComponentManager &_ecm) const
+{
+  return _ecm.EntitiesByComponents(
+      components::ParentEntity(this->dataPtr->id),
+      components::Collision());
+}
+
+//////////////////////////////////////////////////
+std::vector<Entity> Link::Visuals(const EntityComponentManager &_ecm) const
+{
+  return _ecm.EntitiesByComponents(
+      components::ParentEntity(this->dataPtr->id),
+      components::Visual());
+}
+
+//////////////////////////////////////////////////
+uint64_t Link::CollisionCount(const EntityComponentManager &_ecm) const
+{
+  return this->Collisions(_ecm).size();
+}
+
+//////////////////////////////////////////////////
+uint64_t Link::VisualCount(const EntityComponentManager &_ecm) const
+{
+  return this->Visuals(_ecm).size();
+}
+
+//////////////////////////////////////////////////
+bool Link::IsCanonical(const EntityComponentManager &_ecm) const
+{
+  auto comp = _ecm.Component<components::CanonicalLink>(this->dataPtr->id);
+  return comp != nullptr;
+}
+
+//////////////////////////////////////////////////
+bool Link::WindMode(const EntityComponentManager &_ecm) const
+{
+  auto comp = _ecm.Component<components::WindMode>(this->dataPtr->id);
+  if (comp)
+    return comp->Data();
```

Comment:
> we don't immediately know the type of Data() is compatible with the return bool value

After casting the component to `components::WindMode`, yes, the data will always be bool. If `WindMode`'s data type ever changes, this will fail compilation. Note that this is different from `CanonicalLink` above, which doesn't hold any data. 

I guess it may be hard for a developer to know that the type of `WindMode` is `bool` without looking at that class' definition, but I'm not sure how to make this clearer here. 

> Is comp->Data() always going to be true if set, or can it be set as true or false? 

Any system could change the set the value to true or false.